### PR TITLE
Auth GitHub with token instead of the disabled user/pass

### PR DESCRIPTION
**Pull request by michaljurecko**
Jira: https://keboola.atlassian.net/browse/COM-530
Changes:
- GitHub login use token instead of user/password pair (now disabled by GitHub: https://developer.github.com/changes/2020-02-14-deprecating-password-auth/)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -18,11 +18,13 @@ COPY . /init-code/
 WORKDIR /init-code/
 RUN /init-code/composer-install.sh \
 	&& php composer.phar --no-interaction install \
-	&& rm -f composer.phar \ 
+	&& rm -f composer.phar \
 	&& gem install travis \
+	# Install shell completion
+	&& echo "y" | travis \
```
Comment (michaljurecko) [inline Dockerfile:24]: Aby sa to pri spusteni `travis` prikazu nepytalo `Shell completion not installed. Would you like to install it now? |y|`.

File: src/GenerateCommand.php [hunk 2]
```diff
@@ -25,7 +25,8 @@ protected function configure()
             . 'of Keboola Connection application in the chosen language.'
         )
         ->addOption('setup-only', 's', InputOption::VALUE_NONE, 'Only setup deployment')
-        ->addOption('update', 'u', InputOption::VALUE_NONE, 'Update skeleton');
+        ->addOption('update', 'u', InputOption::VALUE_NONE, 'Update skeleton')
+        ->addOption('github-token', 't', InputOption::VALUE_OPTIONAL, 'GitHub API token');
```
Comment (michaljurecko) [inline src/GenerateCommand.php:29]: Token je mozne zadat ako parameter, alebo ho zadat v runtime priamo do konzoly.

File: src/GenerateCommand.php [hunk 3]
```diff
@@ -131,8 +132,12 @@ protected function execute(InputInterface $input, OutputInterface $output)
         }
 
         $output->writeln("Setting up Travis integration.");
+
+        $output->writeln("");
         $output->writeln("Github login");
-        $process = new Process("travis login --pro");
+        $githubToken = $this->getGithubToken($input, $output);
+
+        $process = new Process("travis login --pro --github-token " . escapeshellarg($githubToken));
```
Comment (michaljurecko) [inline src/GenerateCommand.php:140]: Do prikazu `travis login` sa pridava token.

File: src/GenerateCommand.php [hunk 4]
```diff
@@ -185,4 +190,30 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+        // Request token from user
+        $output->writeln("Please provide a GitHub token from the https://github.com/settings/tokens");
+        $output->writeln("For required scopes, see https://docs.travis-ci.com/user/github-oauth-scopes");
+
+        $helper = $this->getHelper('question');
+        $question = new Question('GitHub token:');
+        $githubToken = $helper->ask($input, $output, $question);
+
+        // Remove previous line and replace token with *****
+        if ($output->isDecorated()) {
```
Comment (michaljurecko) [inline src/GenerateCommand.php:212]: Idealne by bolo namiesto skopirovaneho tokenu zobrazovat `***`, no to v Symfony console nejde. Da sa iba uplne skryt vstup, ... no to je pri kopirovani blbe, nevies ci si token skopiroval cely, ci si ho nevlozil 2x, ... kedze po vlozeni sa nic nestane. ..... preto sa token po vlozeni zobrazi, ale po stlaceni ENTER sa nahradi `***` ak to terminal podporuje.

Comment (michaljurecko) [reply src/GenerateCommand.php:212]: Zmazanie riadku je prebrate z Symfony Console ProgressBar: https://github.com/symfony/console/blob/3179d1350843d2caa796973173dd0609ed4c7c67/Helper/ProgressBar.php#L386

**Comment by michaljurecko** [comment]
@ondrajodas ping

**Comment by ondrajodas** [comment]
v pohodě, ještě to jdu vyzkoušet

**Review by ondrajodas — APPROVED** [review]
👍

**Pull request merged**